### PR TITLE
Add class and file filter parameters to Fake.Testing.ReportGenerator

### PR DESCRIPTION
**Pull request by magicmonty**
### Description

The ReportGenerator has added File and class filters. This PR will reflect these change with the addition of two additional optional parameters `ClassFilter` and `FileFilter`

## TODO

- [x] New (API-)documentation for new features exist
- [x] no unit or integration test, doesn't make sense, as it would add a dependency to the external tool
- [x] Fake 5 [API guideline](https://fake.build/contributing.html#API-Design) is honored

### CODE REVIEW

**Comment by magicmonty** [comment]
The failing on CircleCI seems to have nothing to do with this change:
It fails at Fake.DotNet.MSBuild.IntegrationTests

File: src/app/Fake.Testing.ReportGenerator/ReportGenerator.fs [hunk 1]
```diff
@@ -79,15 +93,19 @@ let private ReportGeneratorDefaultParams =
...
-    let filters = sprintf "-filters:%s" (String.Join(";", parameters.Filters))
+    let assemblyFilters = sprintf "-assemblyfilters:%s" (String.Join(";", parameters.Filters))
+    let classFilters = sprintf "-classfilters:%s" (String.Join(";", parameters.ClassFilters))
+    let fileFilters = sprintf "-filefilters:%s" (String.Join(";", parameters.FileFilters))
 
     new StringBuilder()
-    |> StringBuilder.append (sprintf "-reports:%s" (String.Join(";", reports)))
-    |> StringBuilder.append (sprintf "-targetdir:%s" parameters.TargetDir)
+    |> StringBuilder.appendWithoutQuotes (sprintf "-reports:%s" (String.Join(";", reports)))
+    |> StringBuilder.appendWithoutQuotes (sprintf "-targetdir:\"%s\"" parameters.TargetDir)
```
Comment (matthid) [inline src/app/Fake.Testing.ReportGenerator/ReportGenerator.fs]: I'd prefer building a sequence/array and then using `Args.toWindowsCommandLine` which will handle escaping properly (which your quotes are not, for example when `TargetDir` ends with a backslack)

Comment (magicmonty) [reply src/app/Fake.Testing.ReportGenerator/ReportGenerator.fs]: Oh cool, that is one I didn't know of. I will fix it

**Comment by magicmonty** [comment]
@matthid Changed the Argument generation

**Comment by matthid** [comment]
> no unit or integration test, doesn't make sense, as it would add a dependency to the external tool

@magicmonty just to let you know we are discussing this https://github.com/fsharp/FAKE/pull/2114#issuecomment-427788331 I might come up with some idea on how to add some rudimentary unit tests and might ask you to implement some. But if you have an idea feel free to share :)

Otherwise this looks good. I'd assume circleci will go green when merging latest `release/next` (but no need to do so)

**Comment by magicmonty** [comment]
@matthid I'm happy to add tests, but currently I have no Idea how without adding additional dependencies here. I have a little test project to test this stuff out manually but this would add a dependency on ReportGenerator to this project

**Comment by matthid** [comment]
@magicmonty please take a look at the place/test I commented, feel free to ask if you need any help

**Comment by matthid** [comment]
Hm, I tried to merge in latest `release/next` changes into this PR and now apparently github cannot handle the diff anymore :(

**Comment by magicmonty** [comment]
I'll make a rebase on release/next if this is ok

**Comment by matthid** [comment]
Yes, sorry for the inconvenience...

**Comment by magicmonty** [comment]
@matthid Added Unit Tests

**Comment by matthid** [comment]
Thanks!

**Pull request merged**